staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -271,8 +274,8 @@ func finishRequest(timeout time.Duration, fn resultFunc) (result runtime.Object,
 		return nil, err
 	case p := <-panicCh:
 		panic(p)
-	case <-time.After(timeout):
-		return nil, errors.NewTimeoutError(fmt.Sprintf("request did not complete within requested timeout %s", timeout), 0)
+	case <-ctx.Done():
+		return nil, errors.NewTimeoutError(fmt.Sprintf("request did not complete within requested timeout %s", ctx.Err()), 0)
```

Comment:
does `ctx.Err` contain the timeout in seconds?

Reply:
it's not. But we decided it does not matter as the request URL will have the timeout.